java-checks/src/main/java/org/sonar/java/checks/AnonymousClassShouldBeLambdaCheck.java

```diff
@@ -70,20 +75,51 @@ public void visitNewClass(NewClassTree tree) {
     if (classBody != null) {
       TypeTree identifier = tree.identifier();
       if (!useThisIdentifier(classBody) && !enumConstants.contains(identifier) && isSAM(classBody)) {
-        context.reportIssue(this, identifier, "Make this anonymous inner class a lambda" + context.getJavaVersion().java8CompatibilityMessage());
+        Tree member = classBody.members().get(0);
+        if (member.is(Tree.Kind.METHOD) && multipleMethodsDefaultSameSignature(((JavaSymbol.MethodJavaSymbol) ((MethodTree) member).symbol()))) {
+          context.reportIssue(this, identifier, "Make this anonymous inner class a lambda" + context.getJavaVersion().java8CompatibilityMessage());
+        }
       }
     }
   }
 
   private static boolean isSAM(ClassTree classBody) {
-    if(hasOnlyOneMethod(classBody.members()) && classBody.symbol().isTypeSymbol()) {
+    if (hasOnlyOneMethod(classBody.members()) && classBody.symbol().isTypeSymbol()) {
       // Verify class body is a subtype of an interface
       JavaSymbol.TypeJavaSymbol symbol = (JavaSymbol.TypeJavaSymbol) classBody.symbol();
-      return symbol.getInterfaces().size() == 1 && symbol.getSuperclass().is("java.lang.Object");
+      return symbol.getInterfaces().size() == 1 &&
+        symbol.getSuperclass().is("java.lang.Object");
     }
     return false;
   }
 
+  private static boolean multipleMethodsDefaultSameSignature(JavaSymbol.MethodJavaSymbol methodSymbol) {
+    JavaSymbol.TypeJavaSymbol typeJavaSymbol = methodSymbol.overriddenSymbol().enclosingClass();
+    Set<JavaSymbol.MethodJavaSymbol> methods = new HashSet<>();
+    methods.addAll(
+      methodSymbol.overriddenSymbol().enclosingClass().memberSymbols().stream().filter(Symbol::isMethodSymbol).map(x -> ((JavaSymbol.MethodJavaSymbol) x))
+        .collect(Collectors.toList()));
+    List<TypeJavaSymbol> list = typeJavaSymbol.superTypes().stream().filter(type -> !type.name().equals("Object"))
+      .map(ClassJavaType::getSymbol).collect(Collectors.toList());
+    for (JavaSymbol.TypeJavaSymbol t : list) {
+      methods.addAll(t.memberSymbols().stream().filter(Symbol::isMethodSymbol).map(x -> ((JavaSymbol.MethodJavaSymbol) x))
+        .filter(method -> !method.name().equals(methodSymbol.name()))
+        .collect(Collectors.toList()));
+    }
+    return checkForDefaultMethodsWithSameSignature(methodSymbol, methods);
+  }
+
+  private static boolean checkForDefaultMethodsWithSameSignature(JavaSymbol.MethodJavaSymbol methodSymbol, Set<JavaSymbol.MethodJavaSymbol> methods) {
+    List<JavaSymbol.MethodJavaSymbol> list1 = methods.stream().filter(method -> method.getParameters().scopeSymbols()
+      .equals(methodSymbol.getParameters().scopeSymbols()) && !method.name().equals("main") && method.isDefault())
+      .collect(Collectors.toList());
+    if (list1.isEmpty()) {
+      return true;
+    } else {
+      return list1.size() > 1 ? false : true;
```

Comment:
useless ternary operator.  You can replace the whole `if` by `return list1.isEmpty() || list1.size() <= 1`